core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -328,15 +319,89 @@ public long getUsedBytes() {
   }
 
   @Override
-  public List<WorkerInfo> getLostWorkersInfoList() {
-    List<WorkerInfo> ret = new ArrayList<>(mLostWorkers.size());
+  public List<WorkerInfo> getWorkerInfoList() throws UnavailableException {
+    if (mSafeModeManager.isInSafeMode()) {
+      throw new UnavailableException(ExceptionMessage.MASTER_IN_SAFEMODE.getMessage());
+    }
+    List<WorkerInfo> workerInfoList = new ArrayList<>(mWorkers.size());
+    for (MasterWorkerInfo worker : mWorkers) {
+      synchronized (worker) {
+        workerInfoList.add(worker.generateWorkerInfo(null, true));
+      }
+    }
+    return workerInfoList;
+  }
+
+  @Override
+  public List<WorkerInfo> getLostWorkersInfoList() throws UnavailableException {
+    if (mSafeModeManager.isInSafeMode()) {
+      throw new UnavailableException(ExceptionMessage.MASTER_IN_SAFEMODE.getMessage());
+    }
+    List<WorkerInfo> workerInfoList = new ArrayList<>(mLostWorkers.size());
     for (MasterWorkerInfo worker : mLostWorkers) {
       synchronized (worker) {
-        ret.add(worker.generateClientWorkerInfo());
+        workerInfoList.add(worker.generateWorkerInfo(null, false));
       }
     }
-    Collections.sort(ret, new WorkerInfo.LastContactSecComparator());
-    return ret;
+    Collections.sort(workerInfoList, new WorkerInfo.LastContactSecComparator());
+    return workerInfoList;
+  }
+
+  @Override
+  public List<WorkerInfo> getWorkerReport(GetWorkerReportOptions options)
+      throws UnavailableException, InvalidArgumentException {
+    if (mSafeModeManager.isInSafeMode()) {
+      throw new UnavailableException(ExceptionMessage.MASTER_IN_SAFEMODE.getMessage());
+    }
+
+    Set<MasterWorkerInfo> selectedLiveWorkers = null;
```

Comment:
NIT: I would do this to simplify the code:
```java
Set<MasterWorkerInfo> selectedLiveWorkers = new HashSet<>();
Set<MasterWorkerInfo> selectedLostWorkers = new HashSet<>();
...
switch (workerRange) {
  case ALL:
    selectedLiveWorkers.addAll(mWorkers);
    ...
}
for (MasterWorkerInfo worker : selectedLiveWorkers) {
  ....
}
```

Reply:
Done, thanks!